src/renderer/components/+extensions/extensions.tsx

```diff
@@ -172,8 +172,16 @@ async function createTempFilesAndValidate({ fileName, dataP }: InstallRequest, d
 
   // validate packages
   const tempFile = getExtensionPackageTemp(fileName);
-
-  disposer.push(() => fse.unlink(tempFile));
+  let id: string | undefined = undefined;
+
+  disposer.push(() => {
+    // This is necessary so that this file is only removed on the error condition.
+    // if `id` is set then when the disposer is optionally called in `unpackExtension`
+    // this file will not be removed.
+    if (!id) {
+      fse.unlink(tempFile).catch(noop);
```

Comment:
Should we use OS temp dir for these?

Reply:
Probably, would be much easier to deal with...

Reply:
So we already do... shall I just remove this cleanup?